transcendence-app/src/chat/chat.controller.ts

```diff
@@ -139,4 +141,27 @@ export class ChatController {
     }
     return chatroom;
   }
+
+  @Get(':userId/messages')
+  @ApiOkResponse({
+    description: `Get chat messages in a one to one conversation(max ${MAX_ENTRIES_PER_PAGE})`,
+    type: [ChatMessage],
+  })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiUnprocessableEntityResponse({ description: 'Unprocessable entity' })
+  async getChatMessages(
+    @GetUser() userMe: User,
+    @Param('userId', ParseUUIDPipe) recipientId: string,
+    @Query() requestDto: ChatMessagePaginationRequestDto,
+  ) {
+    const messages = await this.chatService.getOneToOneChatMessages(
+      userMe.id,
+      recipientId,
+      requestDto,
+    );
+    if (!messages) {
+      throw new UnprocessableEntityException();
```

Comment:
don't agree with this, we have the unprocessable entity exception in many similar cases. If we get to this point it means we understood the request but didn't find data. The service unavailable is more related to a server error itself

Reply:
If I'm not mistaken, in this case, if the server understood the request and didn't find any data, the `getOneToOneChatMessages` will return an empty array.
When it returns `null`, it means there is a database error.

Reply:
ok, I see your point. We are using then unprocessableEntity wrong in many places because we are always evaluating it against a null value :( I'll update this one on next PR